cmd/tsidp/tsidp.go

```diff
@@ -161,16 +161,18 @@ func main() {
 	} else {
 		srv.serverURL = fmt.Sprintf("https://%s", strings.TrimSuffix(st.Self.DNSName, "."))
 	}
-	if *flagFunnel {
-		f, err := os.Open(funnelClientsFile)
-		if err == nil {
-			srv.funnelClients = make(map[string]*funnelClient)
-			if err := json.NewDecoder(f).Decode(&srv.funnelClients); err != nil {
-				log.Fatalf("could not parse %s: %v", funnelClientsFile, err)
-			}
-		} else if !errors.Is(err, os.ErrNotExist) {
-			log.Fatalf("could not open %s: %v", funnelClientsFile, err)
+
+	// Load funnel clients from disk if they exist, regardless of whether funnel is enabled
+	// This ensures OIDC clients persist across restarts
+	srv.funnelClients = make(map[string]*funnelClient)
```

Comment:
this is the line I meant to comment on. 

should work without this? https://go.dev/play/p/CErYcWxJZRX